### ISSUE TITLE
AllenNLP retreived answer span confidence score #1265

### ISSUE THREAD

**Issue by poulamic23** [allenai/allennlp#2596]
- OS: [Windows 10]
 - Python version: [3.6.5]
 - AllenNLP version: [0.8.2]
 - PyTorch version: (1.0.1 - installed myself)

**Question**
I'm using the wikitables semantic parser of AllenNLP. 
Everything's working fine.
But I need the confidence score for the answer.
#AllenNLP retreived answer span confidence score #1265
[https://github.com/allenai/allennlp/issues/1265#issuecomment-449175028]
@matt-gardner had suggested the above solution.
I tried the following:

Using the predictor.predict_json method for prediction : 
`i = {"table": table, "question": question}

    p = predictor.predict_json(i)`

Added the below two lines of code in wikitables_semantic_parser.py file
`print(torch.max(question_entity_similarity).item())
   print(torch.nn.functional.softmax(torch.max(question_entity_similarity)))`

Output:
11.975749969482422
tensor(1.)

The output is always the same irrespective of the questions and table.
Am i doing the right thing?? If not please suggest.

**Comment by matt-gardner** [comment]
Huh, yeah, it looks like we don't return this.  The confidence score for the answer is the `score` of the best final state.  You'll want to add it somewhere around here: https://github.com/allenai/allennlp/blob/0f7bcf5261b7bb1e7231e172f88cb18210ea51fd/allennlp/models/semantic_parsing/wikitables/wikitables_semantic_parser.py#L651-L666

Feel free to submit a PR to add this.  Also, just FYI, we'll soon be putting up a *much* better and simpler WikiTables parser, that uses a different language to parse into (Chen Liang's language, written in python, instead of lambda-DCS, which we call out to SEMPRE in java for).  I'm not certain how long that will take, but maybe a couple of weeks.  The code is written and the model is trained, we just need to clean up the code and move it over to the main repo.

**Comment by poulamic23** [comment]
![image](https://user-images.githubusercontent.com/48485339/54268462-3d2e3380-45a1-11e9-8b35-8b9caa0f3345.png)

Do you mean to say that the highlighted one is the score we're looking for?

**Comment by matt-gardner** [comment]
Yes.

**Comment by matt-gardner** [comment]
Closing this, feel free to re-open if you have more questions.

**Comment by poulamic23** [comment]
Thank you @matt-gardner for your help. 
Kindly include the confidence score in your next release.
Please guide me on how to raise a request for the same.
